### PR TITLE
378 alarms + telecare year specific tests

### LINKED ISSUE #378: Alarms + Telecare (year specific) Tests

### PR DESCRIPTION
**Pull request by lizihao-anu**
close #378

### CODE REVIEW

File: R/produce_at_sds_episodes_tests.R [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#' @description Produce the test for the Alarm Telecare and SDS all episodes
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_sc_ch_episodes_path()])
+#'
+#' @return a dataframe with a count of each flag.
+#'
+#' @export
+#' @family social care test functions
+produce_at_sds_episodes_tests <- function(data) {
```
Comment (Jennit07) [inline R/produce_at_sds_episodes_tests.R:12]:
```suggestion
produce_sc_at_tests <- function(data) {
```
Can we rename to this please? I was getting confused with having SDS in the name for this function as we should split these and make them specific to the tests

File: R/produce_at_sds_episodes_tests.R [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        .data$dob,
+        .data$postcode,
+        .data$record_keydate1,
+        .data$record_keydate2,
+        .data$sc_latest_submission,
+        .data$year,
+        .data$recid,
+        .data$smrtype,
+        .data$sc_send_lca
+      )
```
Comment (Jennit07) [inline R/produce_at_sds_episodes_tests.R:35]:
```suggestion
    # keep variables for comparison
    dplyr::select(c(.data$valid_chi:.data$West_Lothian))
```
See function: `produce_source_ch_tests` we can have this as a simlar format and select desirable variables. Check this is righ `valid_chi to west lothian` for selection but this reduces the line of code

**Review by Jennit07 — CHANGES_REQUESTED** [review]
Some comments

File: R/produce_at_sds_episodes_tests.R (hunk 2 above)
Comment (Jennit07) [reply R/produce_at_sds_episodes_tests.R:35]: try testing this code for selecting the last variable e.g. west lothian
dplyr::select(c(.data$valid_chi:dplyr::last_col()))

File: R/produce_source_at_tests.R [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      sc_day_care_yes = dplyr::if_else(.data$sc_day_care == 1, 1, 0),
+      sc_day_care_unknown = dplyr::if_else(.data$sc_day_care == 9, 1, 0),
+      sc_housing_mainstream = dplyr::if_else(.data$sc_type_of_housing == 1, 1, 0),
+      sc_housing_supported = dplyr::if_else(.data$sc_type_of_housing == 2, 1, 0),
+      sc_housing_longstaych = dplyr::if_else(.data$sc_type_of_housing == 3, 1, 0),
+      sc_housing_hospital = dplyr::if_else(.data$sc_type_of_housing == 4, 1, 0),
+      sc_housing_other = dplyr::if_else(.data$sc_type_of_housing == 5, 1, 0),
+      sc_housing_unknown = dplyr::if_else(.data$sc_type_of_housing == 6, 1, 0),
+      n_missing_sending_loc = dplyr::if_else(is_missing(.data$sc_send_lca), 1, 0),
+      n_missing_person_id = dplyr::if_else(is_missing(.data$person_id), 1, 0),
```
Comment (Jennit07) [inline R/produce_source_at_tests.R:39]: These variables are not needed here as there should be tests for the client file which contains all of these variables
```suggestion
```

**Comment by lizihao-anu** [comment]
> I've made a few small changes and pushed the commits. Happy to merge

Cool. Merge now

**Pull request merged**